bin/verify_rke2_charts

```diff
@@ -0,0 +1,110 @@
+#!/bin/sh
+
+set -e
+
+. libstd-ecm.sh
+
+# Defaults
+REPOSITORY=rancher/rke2
+LOCAL_RKE2=''
+BRANCH='master'
+IGNORE_LIST=''
+
+usage() {
+    echo "usage: $0 [hbilr]
+    -b    set branch. Default $BRANCH
+    -r    set Github repository name. Default $REPOSITORY
+    -i    add chart ignore list
+    -l    use local repository path. -r is ignored when this is set.
+    -h    show help
+
+examples:
+    $0
+    $0 -i 'rke2-metrics-server rancher-vsphere-cpi' -r my-gh-user/rke2
+    $0 -l ../rke2 -b release-1.22"
+}
+
+while getopts 'b:i:l:r:dh' c; do
+    case "${c}" in
+    b)
+        BRANCH=$OPTARG
+        ;;
+    d)
+        set_debug
+        ;;
+    i)
+        IGNORE_LIST=$OPTARG
+        ;;
+    l)
+        LOCAL_RKE2=$OPTARG
+        ;;
+    r)
+        REPOSITORY=$OPTARG
+        ;;
+    h)
+        usage
+        exit 0
+        ;;
+    *)
+        usage
+        exit 1
+        ;;
+    esac
+done
+
+setup_tmp
+
+# gh environment variable overrides
+export GH_REPO="${REPOSITORY}"
+export PAGER='cat'
+
+# Get required file either locally or remotely
+if [ -z "${LOCAL_RKE2}" ]; then
+    cd "${TMP_DIR}"
+
+    has_gh
+
+    gh api -XGET --field "ref=${BRANCH}" --header "Accept:application/vnd.github.VERSION.raw" \
+        '/repos/{owner}/{repo}/contents/Dockerfile' >Dockerfile
+else
+    cd "${LOCAL_RKE2}"
+
+    if [ "$(git rev-parse --is-shallow-repository)" = 'false' ]; then
+        git checkout -q "${BRANCH}"
```

Comment:
Should we add a step here to pull from upstream to make sure we've got the latest of this branch?

Reply:
I can add it, but bear in mind it will mutate the local environment, potentially delivering "surprises" to the operator. Also there are many cases it would fail like merge/ff conflicts or it does not have the upstream defined locally.

Reply:
yeah, let's defer that for now.